chore(cli): add a method on AuthenticatedClient to get user info

y-lakhdar opened the pull request:
Just to prevent redoing the same logic on ui:create commands

louis-bompart commented on packages/cli/src/lib/platform/authenticatedClient.ts:
```diff
@@ -65,6 +65,17 @@ export class AuthenticatedClient {
     });
   }
+
+  async platformUserCredentials(name: string) {
+    const authenticatedClient = new AuthenticatedClient();
+    const platformClient = await authenticatedClient.getClient();
+    await platformClient.initialize();
+
+    const userInfo = await platformClient.user.get();
+    const apiKey = await authenticatedClient.createImpersonateApiKey(name);
+
+    return {userInfo, apiKey};
+  }
```
Do you think this UT-able? it'd be great tbh

louis-bompart reviewed (commented):
Can you write UT for the method while at it?
That'd make the code even more reliable :D

y-lakhdar commented:
I changed the function so it only does one thing : return the user info. 
I think it was weird for the function to return the user info and also create an API ([commit](https://github.com/coveo/cli/pull/176/commits/77e964e10be44b49d2bfc7e71a7d0b0add4f9d15)).

Also added UT

olamothe commented on packages/cli/src/lib/platform/authenticatedClient.ts:
```diff
@@ -65,6 +65,14 @@ export class AuthenticatedClient {
+    const platformClient = await authenticatedClient.getClient();
+    await platformClient.initialize();
+
+    return await platformClient.user.get();
```
the `await` is not necessary here since we return immediately anyway.

olamothe commented on packages/cli/src/lib/platform/authenticatedClient.ts:
```diff
@@ -65,6 +65,14 @@ export class AuthenticatedClient {
+  async getUserInfo() {
+    const authenticatedClient = new AuthenticatedClient();
+    const platformClient = await authenticatedClient.getClient();
+    await platformClient.initialize();
```
Why do we need to initialize here ?

y-lakhdar replied:
Because, if I don't pass any argument to the `user.get()` method, it tries to load the currentUser (which does not exist at that moment)
https://github.com/coveo/platform-client/blob/master/src/resources/Users/User.ts#L8

olamothe commented on packages/cli/src/lib/platform/authenticatedClient.spec.ts:
```diff
@@ -121,4 +127,10 @@ describe('AuthenticatedClient', () => {
   });
+
+  it('#getUserInfo should return the user info', async () => {
+    const userInfo = await new AuthenticatedClient().getUserInfo();
```
Related to my comment below, if `initialize` is needed on the platform client in order to correctly receive the user info, then this is what the UT should focus on.

Because, otherwise, this test is kind of useless, we mock the return and just check that it returns what we just mocked ;)
